packages/core/src/components/tree-view/tree-view.tsx

```diff
@@ -0,0 +1,214 @@
+import { h, Component, ComponentInterface, Event, EventEmitter, Prop } from '@stencil/core';
+
+import { TreeViewItem } from '@dso-toolkit/sources';
+import { DsoTreeItem } from './tree-item';
+
+@Component({
+  tag: 'dso-tree-view',
+  styleUrl: './tree-view.scss',
+  shadow: true
+})
+export class TreeView implements ComponentInterface {
+   /**
+   * The collection of TreeViewItems
+   */
+  @Prop()
+  collection!: TreeViewItem<string>[];
+
+  /**
+   * Emitted when a tree view item is opened.
+   * The `detail` property of the `CustomEvent` will contain the complete path of TreeViewItems from the
+   * root to the item that is emitting the open event. The consumer of the event is responsible for updating
+   * the TreeView's collection (usually set the open state on the last TreeViewItem in path).
+   */
+  @Event()
+  openItem!: EventEmitter<TreeViewItem<string>[]>;
+
+  /**
+   * Emitted when a tree view item is closed.
+   * The `detail` property of the `CustomEvent` will contain the complete path of TreeViewItems from the
+   * root to the item that is emitting the close event. The consumer of the event is responsible for updating
+   * the TreeView's collection (usually set the closed state on the last TreeViewItem in path).
+   */
+  @Event()
+  closeItem!: EventEmitter<TreeViewItem<string>[]>;
+
+  /**
+   * Emitted when a tree view item is clicked.
+   * The `detail` property of the `CustomEvent` will contain the complete path of TreeViewItems from the
+   * root to the item that is emitting the clicked event.
+   */
+  @Event()
+  clickItem!: EventEmitter<TreeViewItem<string>[]>;
+
+  keyDownListener = (event: KeyboardEvent) => {
```

Comment:
Uitgebreid voorstel dus blijf bij me ;-)

* Deze property function omschrijven naar een instance method
* De `onKeyDown` method listener vervangen met de `@Listen('onKeyDown')` decorator van Stencil: https://stenciljs.com/docs/events#listen-decorator
* Een property `@Element() element: HTMLElement` toevoegen
* Op regel 51 geen `.find()` maar `.some(item => item === this.element);`
* Alle static methods omschrijven naar instance methods die dan `this.element.shadowRoot().xxx()` kunnen gebruiken.

Je kan ipv. `@Element()` ook een DOM node reference naar `.tree-view` maken: https://stenciljs.com/docs/templating-jsx#getting-a-reference-to-a-dom-element Dan werk je niet met `this.element.shadowRoot()` maar met `this.treeViewElement` (oid)

Ik denk dat de code leesbaarder wordt:
* Native Stencil functionaliteit om het host element te vinden (ipv
* Geen element _zoeken_ maar _hebben_.

maar het is up to you., dit werkt ook en waarschijnlijk moeten we toch nog ff doorsleutelen waardoor we dit ook later kunnen oppakken